packages/@react-aria/overlays/src/useOverlayPosition.ts

```diff
@@ -171,6 +171,9 @@ export function useOverlayPosition(props: AriaPositionProps): PositionAria {
     // This is so we don't have to delay autoFocus scrolling or delay applying preventScroll for popovers
     overlay.style.top = '';
     overlay.style.bottom = '';
+    overlay.style.left = '';
+    overlay.style.right = '';
+
```

Comment:
When the tooltip is flipped from its previously calculate position, there is a scenario where both left and right are applied on the overlay at the same time where the left/right set respectively are the values to position the tooltip to the right/left of the button respectively, resulting in the tooltip squishing itself in an attempt to fufill both constraints. Clearing it out here means this won't happen